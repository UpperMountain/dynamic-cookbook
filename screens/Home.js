import React from "react";
import {
  View,
  StyleSheet,
  TouchableWithoutFeedback,
  Keyboard
} from "react-native";
import { LinearGradient } from "expo";
import SearchBar from "../components/SearchBar";
import SafeView from "../components/SafeView";
import { MealCard } from "../components/MealCard";
import { FontAwesome } from "@expo/vector-icons";
import tabIcon from "../components/tabIcon";

class Home extends React.Component {
  constructor(props) {
    super();
    this.state = {
      searching: false
    };
  }

  static navigationOptions = {
    tabBarIcon: tabIcon("home")
  };

  render() {
    const DismissKeys = ({ children }) => (
      <TouchableWithoutFeedback
        onPress={() => {
          Keyboard.dismiss();
        }}
      >
        {children}
      </TouchableWithoutFeedback>
    );

    return (
      <DismissKeys>
        <LinearGradient
          style={styles.container}
          colors={["#FFAFBD", "#FFC3A0"]}
        >
          <SafeView>
            <SearchBar open={this.state.searching} />
          </SafeView>
        </LinearGradient>
      </DismissKeys>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    display: "flex"
  },
  background: {
    flex: 1,
    display: "flex"
  }
});

export default Home;
