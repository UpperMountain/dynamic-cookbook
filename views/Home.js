import React from "react";
import {
  View,
  StyleSheet,
  TouchableWithoutFeedback,
  Keyboard,
  StatusBar
} from "react-native";
import { LinearGradient } from "expo";
import MenuBar from "../components/MenuBar";
import SearchBar from "../components/SearchBar";
import { List } from "../components/RecipeList";
import { MealCard } from "../components/MealCard";

class Home extends React.Component {
  constructor(props) {
    super();
    this.state = {
      searching: false
    };
  }

  render() {
    const DismissKeys = ({ children }) => (
      <TouchableWithoutFeedback
        onPress={() => {
          Keyboard.dismiss();
        }}
      >
        {children}
      </TouchableWithoutFeedback>
    );

    return (
      <DismissKeys>
        <View style={styles.container}>
          <StatusBar barStyle="light-content" />
          <LinearGradient
            style={styles.background}
            colors={["#FFAFBD", "#FFC3A0"]}
          >
            <SearchBar open={this.state.searching} />
            <MenuBar />
          </LinearGradient>
        </View>
      </DismissKeys>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "transparent"
  },
  background: {
    flex: 1,
    alignItems: "center"
  }
});

export default Home;
