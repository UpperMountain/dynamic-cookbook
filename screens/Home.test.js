import React from "react";
import renderer from "react-test-renderer";
import Home from "./Home";

it("renders without exploding", () => {
  renderer.create(<Home />);
});

it("passes snapshot tests", () => {
  const cmp = renderer.create(<Home />);
  expect(cmp.toJSON()).toMatchSnapshot();
});
