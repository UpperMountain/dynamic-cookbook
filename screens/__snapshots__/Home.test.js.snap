// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`passes snapshot tests 1`] = `
<ExponentLinearGradient
  accessible={true}
  colors={
    Array [
      4294946749,
      4294951840,
    ]
  }
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "display": "flex",
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    style={
      Object {
        "flex": 1,
        "paddingTop": 0,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(255, 255, 255, .2)",
          "borderRadius": 72,
          "display": "flex",
          "flexDirection": "row",
          "height": 56,
          "margin": 20,
          "paddingLeft": 36,
          "paddingRight": 36,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Search Recipes..."
        placeholderTextColor="white"
        style={
          Object {
            "color": "white",
            "flex": 1,
            "fontFamily": "raleway",
            "fontSize": 18,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <Text />
    </View>
  </RCTSafeAreaView>
</ExponentLinearGradient>
`;
